app/controllers/cinema_halls_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class CinemaHallsController < ApplicationController
+  def index
+    @cinema_halls = CinemaHall.all
+    render json: @cinema_halls
+  end
+
+  def show
+    cinema_hall = CinemaHall.find(params[:id])
+    render json: cinema_hall
+  end
+
+  def create
+    @cinema_hall = CinemaHall.new(cinema_hall_params)
+    if [20, 50, 100, 200].include? @cinema_hall.size
+      @cinema_hall.save
+      cinema_hall_rows_columns
```

Comment:
You do not want to have empty `column_num` and `row_num` columns in your `cinema_hall` table. It would be more readable if you first assign `size` `name` and also `column_num` and `row_num` to a record and then check if it is saved. 

Anyway if creating a record is success you should return the record but you do not return anything.